### PR TITLE
Added more definition tests and more example parsing

### PR DESCRIPTION
**Pull request by Razzeee**
@jmbockhorst 

Please have a look, this is doing some of the parsing we need for multiple files in one inline string.

### CODE REVIEW

File: src/test/definitionProviderTests/definitionProviderTestBase.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            this.treeParser.getWorkspace(determinedTestType.sources),
+          );
+
+          expect(definition).toEqual(undefined);
+        }
+        break;
+
+      case "resolvesToDifferentFile":
+        {
+          const definition = this.definitionProvider.handleDefinition(
```
Comment (Razzeee) [inline src/test/definitionProviderTests/definitionProviderTestBase.ts:55]: This branch is not used yet and needs changing

File: src/test/utils/sourceParser.ts [hunk 2]
```diff
@@ -2,63 +2,137 @@ import { Position } from "vscode-languageserver";
...
-      result.position = { line, character };
+      tempPosition = { line, character };
     }
   });
 
+  if (!tempPosition) {
+    fail();
+  }
+
+  tempSource = ["module Test exposing (..)", "", ...tempSource];
```
Comment (Razzeee) [inline src/test/utils/sourceParser.ts:22]: I think we should remove this and do it explicitly in every test string

File: src/test/utils/sourceParser.ts [hunk 3]
```diff
@@ -2,63 +2,137 @@ import { Position } from "vscode-languageserver";
...
+
+  (tempPosition as Position).line += 2;
+
+  const result: {
+    newSources: { [K: string]: string };
+    position: Position;
+  } = { newSources: {}, position: tempPosition };
+  result.newSources["Main.elm"] = tempSource.join("\n");
```
Comment (Razzeee) [inline src/test/utils/sourceParser.ts:30]: This needs to change to handle multiple files. But it should be easy from here.

**Comment by jmbockhorst** [comment]
I was actually in the middle of working on it too, but I wasn't too far into yet.

File: src/test/utils/sourceParser.ts (hunk 2 above)
Comment (jmbockhorst) [reply src/test/utils/sourceParser.ts:22]: Yeah, I was thinking that too.

**Comment by Razzeee** [comment]
Thank god, sorry for doing this unannounced. :)

**Comment by Razzeee** [comment]
I would be willing to merge it as is, if you agree on the direction we're going. 

Will do some follow ups in the next days to ignore the failing tests and add the missing handling for jump to different file tests.

You could work on the completions side of things then.

**Comment by jmbockhorst** [comment]
I'll wait until this is merged and pull it into the other branch and use it to parse multiple files.

**Pull request merged**